### PR TITLE
Refactor person to student

### PR DESCRIPTION
**Pull request by gnahzeus**
Improves code quality by refactoring person to student in 
- Class names ✅ 
- method names ✅
- json files ✅
- puml files ✅
- docs ✅
- PERSON constants are not changed. ❌ 
- Person in UI not changed. ❌ 

Also fixes style/spelling errors from v1.3 closing.

Addresses #216

### CODE REVIEW

**Review by Echomo-Xinyu — CHANGES_REQUESTED** [review]
Thanks for taking up this tedious work! I have checked your branch and it seems that there are still a few more `Person`s present in the repository. I shall not list down further as you can do a `control+shift+F` to do a global search in the project to catch the missed ones.

**Comment by Echomo-Xinyu** [comment]
Also, remember to export the PlantUML diagrams to png files to the correct folder after you have modified the puml file! It seems like no png has been updated in this PR yet.

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public EditCommand parse(String args) throws ParseException {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
         }
 
-        EditPersonDescriptor editPersonDescriptor = new EditPersonDescriptor();
+        EditStudentDescriptor editPersonDescriptor = new EditStudentDescriptor();
```
[the remark this answers is not in the document]
Comment (gnahzeus) [reply src/main/java/seedu/address/logic/parser/EditCommandParser.java:53]: Thanks for the catch!

File: docs/diagrams/SortBySequenceDiagram.puml [hunk 2]
```diff
@@ -10,7 +10,7 @@ end box
 box Model MODEL_COLOR_T1
 participant ":Model" as Model MODEL_COLOR
 participant ":TeachersPet" as TeachersPet MODEL_COLOR
-participant "persons:UniquePersonList" as UniquePersonList MODEL_COLOR
+participant "persons:UniqueStudentList" as UniqueStudentList MODEL_COLOR
```
[the remark this answers is not in the document]
Comment (gnahzeus) [reply docs/diagrams/SortBySequenceDiagram.puml:13]: Good catch! Will do a global search for the rest of persons.

**Comment by gnahzeus** [comment]
About the PlantUML diagrams to png files, I noticed that there are some other errors (related to AddressBook not changed to Teacher's Pet). I think it is better to address this in another PR that is dedicated to updating the png and puml files for both Person -> Student and AddressBook -> Teacher's Pet. 

@Echomo-Xinyu What are your opinions on this?

**Comment by seanflyyy** [comment]
@gnahzeus what kind of errors did you notice? is it just pertaining to updating from address book to teacher's pet?

**Comment by gnahzeus** [comment]
Yes @seanflyyy, the puml and png files are not updated for Teacher's Pet as well.

**Comment by seanflyyy** [comment]
@gnahzeus sure,in which case let's just modify it in another PR. i'll take it up

**Review by seanflyyy — APPROVED** [review]
LGTM

**Review by Echomo-Xinyu — APPROVED** [review]
LGTM! remember to refactor the remaining two in the future PR!

**Pull request merged**